src/builder/commands/composer.rs

```diff
@@ -0,0 +1,252 @@
+        Some(ref p) if p == "source" => { cmd.arg("--prefer-source"); },
+        Some(ref p) => return Err(From::from(format!(
+            "Value of 'ComposerDependencies.prefer' must be either \
+            'source' or 'dist', '{}' given", p
+        ))),
+        _ => {}
+    }
+
+    run(cmd)
+}
+
+pub fn configure(ctx: &mut Context) -> Result<(), String> {
+    try!(ctx.add_cache_dir(Path::new("/tmp/composer-cache"),
+                           "composer-cache".to_string()));
+
+    ctx.environ.insert("COMPOSER_HOME".to_owned(),
+                       COMPOSER_HOME.to_owned());
+    ctx.environ.insert("COMPOSER_VENDOR_DIR".to_owned(),
+                       COMPOSER_VENDOR_DIR.to_owned());
+    ctx.environ.insert("COMPOSER_BIN_DIR".to_owned(),
+                       COMPOSER_BIN_DIR.to_owned());
+    ctx.environ.insert("COMPOSER_CACHE_DIR".to_owned(),
+                       COMPOSER_CACHE.to_owned());
+
+    Ok(())
+}
+
+pub fn bootstrap(ctx: &mut Context) -> Result<(), String> {
+    try_msg!(create_dir(COMPOSER_HOME, true),
+        "Error creating composer home dir {d:?}: {err}", d=COMPOSER_HOME);
+
+    let composer_inst = try!(download::download_file(ctx, COMPOSER_BOOTSTRAP));
+    try!(copy(&composer_inst, &Path::new("/vagga/root/tmp/composer-setup.php"))
+        .map_err(|e| format!("Error copying composer installer: {}", e)));
+
+    let runtime_exe = ctx.composer_settings
+        .runtime_exe
+        .clone()
+        .unwrap_or(DEFAULT_RUNTIME.to_owned());
+
+    let args = [
+        "/usr/bin/env".to_owned(),
+        runtime_exe,
+        "/tmp/composer-setup.php".to_owned(),
+        "--install-dir=/tmp/".to_owned(),
+    ];
+    try!(run_command(ctx, &args));
+
+    let args = [
+        "ln".to_owned(),
+        "-s".to_owned(),
+        "/tmp/composer.phar".to_owned(),
+        "/usr/local/bin/composer".to_owned(),
+    ];
+    try!(run_command(ctx, &args));
+
+    if ctx.composer_settings.install_runtime {
+        try!(setup_include_path(ctx));
+    }
+
+    Ok(())
+}
+
+fn setup_include_path(ctx: &mut Context) -> Result<(), String> {
+    let args = [
+        "/bin/sh".to_owned(),
+        "-exc".to_owned(),
+        "find $(ls -d /etc/php*) -name 'conf.d' | grep -E '^/etc/php'".to_owned(),
+    ];
+
+    let conf_d = try!(capture_command(ctx, &args, &[])
+        .and_then(|result| {
+            String::from_utf8(result).map_err(|e| format!("{}", e))
+        }));
+
+    let conf_d: Vec<String> = {
+        let conf_d_lines: Vec<String> = conf_d
+            .lines()
+            .filter(|l| !l.is_empty())
+            .map(|l| l.to_owned())
+            .collect();
+
+        if !conf_d_lines.is_empty() {
+            conf_d_lines
+        } else {
+            let args = [
+                "/bin/sh".to_owned(),
+                "-exc".to_owned(),
+                "php --ini | grep -E 'conf.d$' | cut -d ':' -f 2 | cut -d ' ' -f 2".to_owned()
```

Comment:
We don't do shell pipelines in vagga. We assume that there might be no shell in container... Just capture and parse it by rust. It wouldn't be too hard, I swear :)